mono/metadata/class.c

```diff
@@ -289,31 +289,48 @@ mono_type_name_check_byref (MonoType *type, GString *str)
 		g_string_append_c (str, '&');
 }
 
+static char*
+escape_special_chars (const char* identifier)
+{
+	// Assume the worst case, and thus only allocate once
+	char *res = g_malloc (strlen (identifier) * 2 + 1);
```

Comment:
Please store the length in a local.

Reply:
Consider making two passes to save memory.

Reply:
Yeah, wasn't sure on this. I opted to just allocate aggressively since that's what the eglib string functions seemed to do.